core/base-service/deprecated-service.js

```diff
@@ -1,19 +1,40 @@
 'use strict'
 
 const Joi = require('joi')
+const camelcase = require('camelcase')
 const BaseService = require('./base')
+const { isValidCategory } = require('./categories')
 const { Deprecated } = require('./errors')
+const { isValidRoute } = require('./route')
+
+const attrSchema = Joi.object({
+  route: isValidRoute,
+  name: Joi.string(),
+  label: Joi.string(),
+  category: isValidCategory,
+  // The content of examples is validated later, via `transformExamples()`.
+  examples: Joi.array().default([]),
+  message: Joi.string(),
+  dateAdded: Joi.date().required(),
+}).required()
+
+function deprecatedService(attrs) {
```

Comment:
I kinda liked the old function signature with the named fields on the param object. Any reason we can't (or shouldn't) keep that? I guess that'd require a different argument on line 23 below (although I assume something like `arguments[0]` or something else could still make that work)

Not a blocking issue for me but wanted to ask

Reply:
I tend to think of `arguments` as being magical and leading to unexpected things. Though here I guess the reason would be to allow using a more readable function signature with a concise call to `Joi.validate`. I'll do that if you think it improves things.

Reply:
Yeah and that's a fair tradeoff. I just like it when my editor gives me good intellisense 😄 I don't feel particularly strongly either way and so defer to your preference. In fairness, this isn't something that will be used all that commonly (at least I hope there won't be a ton of deprecated services)

Reply:
Hrm, I just went to update this, but ran into an issue where the variables also have to be re-assigned inside if we want to let Joi set default values. I feel like that makes it not worth it.

Related aside: I'm not sure there is an issue open for it yet, but autocomplete-friendly docs for core + helpers is on the roadmap. Maybe we should get that started! Do you know of any tooling for checking that, or what format should be used?

Reply:
> I feel like that makes it not worth it.

SGTM! Doing one final glance over this and then will approve

> Do you know of any tooling for checking that, or what format should be used?

Do you mean tooling to check for autocomplete friendliness? I suppose the answer is no either way 😆 but can start digging into it 👍

Reply:
Yea, I guess it's to validate that the documentation is complete, formatted correctly, and matches the function.

Ooh, there's an eslint plugin!

https://github.com/gajus/eslint-plugin-jsdoc

We could turn it some of those on as warnings, and maybe some as errors, and start working our way through…

Reply:
Nice! We'd able to leverage the eslint overrides feature from the root config to set the rule for certain files/directories progressively to do just that